lib/common/asar.js

```diff
@@ -198,24 +198,32 @@
     }
 
     if (old[util.promisify.custom]) {
-      module[name][util.promisify.custom] = function () {
-        const pathArgument = arguments[pathArgumentIndex]
-        const { isAsar, asarPath, filePath } = splitPath(pathArgument)
-        if (!isAsar) return old[util.promisify.custom].apply(this, arguments)
-
-        const archive = getOrCreateArchive(asarPath)
-        if (!archive) {
-          return Promise.reject(createError(AsarError.INVALID_ARCHIVE, { asarPath }))
-        }
+      module[name][util.promisify.custom] = makePromiseFunction(old[util.promisify.custom], pathArgumentIndex)
+    }
 
-        const newPath = archive.copyFileOut(filePath)
-        if (!newPath) {
-          return Promise.reject(createError(AsarError.NOT_FOUND, { asarPath, filePath }))
-        }
+    if (module.promises && module.promises[name]) {
+      module.promises[name] = makePromiseFunction(module.promises[name], pathArgumentIndex)
+    }
+  }
+
+  const makePromiseFunction = function (orig, pathArgumentIndex) {
+    return function () {
```

Comment:
Can we use `...args` here instead of `arguments` for readability

Reply:
we use `arguments` everywhere here, I can create a follow up PR to refactor everything consistently. There is also some duplicate code that can be merged.

Reply:
Would suggest either a hold on the refactor till this file is TS'ified or TS'ify this file.

Reply:
did the ...args change here